src/OrchardCore.Modules/OrchardCore.Users/Recipes/UsersStep.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using OrchardCore.Recipes.Models;
+using OrchardCore.Recipes.Services;
+using OrchardCore.Sms;
+using OrchardCore.Users.Models;
+using YesSql;
+
+namespace OrchardCore.Users.Recipes;
+
+public class UsersStep : IRecipeStepHandler
+{
+    private readonly UserManager<IUser> _userManager;
+    private readonly ISession _session;
+
+    public UsersStep(
+        UserManager<IUser> userManager,
+        ISession session)
+    {
+        _userManager = userManager;
+        _session = session;
+    }
+
+    public async Task ExecuteAsync(RecipeExecutionContext context)
+    {
+        if (!string.Equals(context.Name, "Users", StringComparison.OrdinalIgnoreCase))
+        {
+            return;
+        }
+
+        var model = context.Step.ToObject<UsersStepModel>();
+
+        foreach (var importedUser in model.Users)
+        {
+            if (string.IsNullOrWhiteSpace(importedUser.UserName))
+            {
+                continue;
+            }
+
+            var iUser = await _userManager.FindByIdAsync(importedUser.UserId);
+
+            if (iUser is not User user)
+            {
+                user = new User { UserId = importedUser.UserId };
+            }
+            else
```

Comment:
The else can be remove. At line 48 `user` property will never be null and will always be an instance of `User`